app/code/core/Mage/Index/Model/Observer.php

```diff
@@ -136,4 +139,35 @@ public function processConfigDataSave(Varien_Event_Observer $observer)
             Mage_Index_Model_Event::TYPE_SAVE
         );
     }
+
+    /**
+     * Clean old index events for indexers in manual mode
+     *
+     * @return void
+     */
+    public function cleanOutdatedEvents()
+    {
+        $manualIndexProcessCollection = Mage::getSingleton('index/indexer')
+            ->getProcessesCollection()
+            ->addFieldToFilter('mode', Mage_Index_Model_Process::MODE_MANUAL);
+
+        $now = new DateTime();
+        /** @noinspection PhpUnhandledExceptionInspection */
+        $dateInterval = new DateInterval('PT' . self::OLD_INDEX_EVENT_THRESHOLD_SECONDS . 'S');
+        $oldEventsThreshold = $now
+            ->sub($dateInterval)
+            ->format('Y-m-d H:i:s');
```

Comment:
```suggestion
            ->format(Varien_Db_Adapter_Pdo_Mysql::TIMESTAMP_FORMAT);
```
I didn't use `Varien_Date::DATETIME_PHP_FORMAT` which also contains this format, because you are using it clearly in a DB context here. I would like to suggest to avoid string literals where constants do exist. I found 18 usages of this literal already (can be addressed in a future PR), let's avoid adding another one 🙂.